src/java.desktop/share/classes/javax/swing/JSplitPane.java

```diff
@@ -989,25 +989,23 @@ protected void addImpl(Component comp, Object constraints, int index)
         }
 
         /* Find the Component that already exists and remove it. */
-        if (constraints != null && (constraints.equals(JSplitPane.LEFT) ||
-                                   constraints.equals(JSplitPane.TOP))) {
+        if (JSplitPane.LEFT.equals(constraints) ||
+            JSplitPane.TOP.equals(constraints)) {
             toRemove = getLeftComponent();
             if (toRemove != null) {
                 remove(toRemove);
             }
             leftComponent = comp;
             index = -1;
-        } else if (constraints != null &&
-                   (constraints.equals(JSplitPane.RIGHT) ||
-                    constraints.equals(JSplitPane.BOTTOM))) {
+        } else if (JSplitPane.RIGHT.equals(constraints) ||
```

Comment:
I do not think that calling the equals on some object is equivalent to calling the equals on String. Please check other cases for similar issues.